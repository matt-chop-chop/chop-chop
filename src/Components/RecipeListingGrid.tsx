import { Grid } from "@chakra-ui/react";
import { RandomRecipeListing, RecipeListingFromData } from "@/Components";

type RecipeListingGridProps = {
  numberOfListings?: number;
  recipes?: string[];
};

const RecipeListingGrid = ({
  numberOfListings = 12,
  recipes = [],
}: RecipeListingGridProps) => {
  console.log(recipes);
  const listings = Array.from(Array(numberOfListings).keys());

  return (
    <Grid
      columnGap={[0, 8, 6]}
      gridAutoRows="1fr"
      gridTemplateColumns={[
        "repeat(1, 1fr)",
        "repeat(2, 1fr)",
        "repeat(3, 1fr)",
      ]}
      mt={[6, 8, 8]}
      rowGap={[6, 8, 8]}
    >
      {listings.map((_, index) => {
        if (recipes.length > 0) {
          return <RecipeListingFromData id={recipes[index]} key={index} />;
        }

        return <RandomRecipeListing key={index} listingIndex={index} />;
      })}
    </Grid>
  );
};
export default RecipeListingGrid;
