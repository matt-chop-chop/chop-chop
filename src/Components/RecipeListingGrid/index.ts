export { default as RecipeListingGrid } from "./RecipeListingGrid";
