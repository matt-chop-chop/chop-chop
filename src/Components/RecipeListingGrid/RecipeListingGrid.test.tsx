import { shallow } from "enzyme";
import React from "react";
import RecipeListingGrid, { RecipeListingGridProps } from "./RecipeListingGrid";

const props: RecipeListingGridProps = {
  showRandom: false,
  recipes: ["123", "456", "789"],
};

const propsRandom: RecipeListingGridProps = {
  showRandom: true,
};

describe("RecipeListingGrid", () => {
  it("is exported from RecipeListingGrid", () => {
    expect(RecipeListingGrid).toBeDefined();
  });

  it("renders as expected when showRandom is false", () => {
    const wrapper = shallow(<RecipeListingGrid {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected when showRandom is true", () => {
    const wrapper = shallow(<RecipeListingGrid {...propsRandom} />);

    expect(wrapper).toMatchSnapshot();
  });
});
