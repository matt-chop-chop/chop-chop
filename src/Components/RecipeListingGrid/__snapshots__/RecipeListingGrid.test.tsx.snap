// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeListingGrid renders as expected when showRandom is false 1`] = `
<Grid
  columnGap={
    [
      0,
      8,
      6,
    ]
  }
  gridAutoRows="1fr"
  gridTemplateColumns={
    [
      "repeat(1, 1fr)",
      "repeat(2, 1fr)",
      "repeat(3, 1fr)",
    ]
  }
  mt={
    [
      6,
      8,
      8,
    ]
  }
  rowGap={
    [
      6,
      8,
      8,
    ]
  }
>
  <RecipeListingFromData
    id="123"
    key="0"
  />
  <RecipeListingFromData
    id="456"
    key="1"
  />
  <RecipeListingFromData
    id="789"
    key="2"
  />
</Grid>
`;

exports[`RecipeListingGrid renders as expected when showRandom is true 1`] = `
<Grid
  columnGap={
    [
      0,
      8,
      6,
    ]
  }
  gridAutoRows="1fr"
  gridTemplateColumns={
    [
      "repeat(1, 1fr)",
      "repeat(2, 1fr)",
      "repeat(3, 1fr)",
    ]
  }
  mt={
    [
      6,
      8,
      8,
    ]
  }
  rowGap={
    [
      6,
      8,
      8,
    ]
  }
>
  <RandomRecipeListing
    key="0"
    listingIndex={0}
  />
  <RandomRecipeListing
    key="1"
    listingIndex={1}
  />
  <RandomRecipeListing
    key="2"
    listingIndex={2}
  />
  <RandomRecipeListing
    key="3"
    listingIndex={3}
  />
  <RandomRecipeListing
    key="4"
    listingIndex={4}
  />
  <RandomRecipeListing
    key="5"
    listingIndex={5}
  />
  <RandomRecipeListing
    key="6"
    listingIndex={6}
  />
  <RandomRecipeListing
    key="7"
    listingIndex={7}
  />
  <RandomRecipeListing
    key="8"
    listingIndex={8}
  />
  <RandomRecipeListing
    key="9"
    listingIndex={9}
  />
  <RandomRecipeListing
    key="10"
    listingIndex={10}
  />
  <RandomRecipeListing
    key="11"
    listingIndex={11}
  />
</Grid>
`;
